New_Authenticator.ipynb

```diff
@@ -0,0 +1,74 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "from jupyterhub.auth import Authenticator\n",
+    "\n",
+    "from tornado import gen\n",
+    "\n",
+    "import csv"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   
+   "outputs": [],
+   "source": [
+    "class MyNewAuthenticator(Authenticator):\n",
+    "        \n",
+    "        \n",
+    "        \n",
+    "\n",
+    "    @gen.coroutine\n",
+    "    def authenticate(self, handler,data):\n",
+    "       with open('Example.csv') as csvfile:\n",
+    "        readCSV=csv.reader(csvfile,delimiter=',')\n",
+    "        \n",
+    "        username[]\n",
+    "        password[]\n",
+    "        for row in readCSV:\n",
+    "            username.append(row[0])\n",
+    "            password.append(row[1])\n",
+    "        len=len(username)\n",
+    "        while(len--):\n",
```

Comment:
In the notebook, try calling this method and seeing what it does. This will not check usernames, it will only check one character at a time.

Additionally, `len--` is not valid Python syntax. Try running this code interactively. Since you are already in a notebook, you should be able to do this in another cell.